src/client/src/core/util/validation/rules.ts

```diff
@@ -636,3 +636,37 @@ export const validLangString = function(value: string) {
 export const validSelection = function(value: string | object | null) {
   return typeof value === 'string' ? translate('general.invalid') : true;
 };
+
+export const validHostnameFormat = function(value: string): boolean | string {
```

Comment:
Not sure whether you considered, [Internationalized domain name](https://en.wikipedia.org/wiki/Internationalized_domain_name), [Punycode](https://en.wikipedia.org/wiki/Punycode)

E.g.
https://github.com/orangehrm/orangehrm-os-mobile/blob/master/src/lib/helpers/url.ts#L27
https://github.com/symfony/symfony/blob/6.1/src/Symfony/Component/Validator/Constraints/UrlValidator.php#L24-L46